### ISSUE TITLE
Make creating camel unit test easier

### ISSUE THREAD

**Issue by fharms** [fabric8io/fabric8-forge#199]
Proposal for adding a new feature to the camel forge plugin to make creating camel unit test easier.

From my work with camel most of the code I have written is actually test code, and I have use a vast amount of time creating the test skeleton for each test. Therefore it could be live saving to have a number of commands in forge to create camel unit tests “out of the box” :-)

To begin with it could create the unit test skeleton based on the camel version and testing mechanisms described on the camel testing guide [Camel Testing](http://camel.apache.org/testing.html). 

But the real value comes when you can quickly build or extend the unit test while working on it. From top of my head I can come up with following other useful use cases.

Other use cases :
- Add adviceWith dialog, select the route and it generate the adviceWith code.
- Add mock dialog, select the route and it generate a general test case with basic code for testing with mock.
- Create unit test skeleton based on an existing route. 
- Add NotifyBuilder dialog, select route(optional) and the various possible expression

**Comment by davsclaus** [comment]
Yeah good ideas with the specialized use-cases for advice / notifier etc.

**Comment by apupier** [comment]
In Fuse Tooling, creating the unit test skeleton based on existing route has been implemented: see https://access.redhat.com/documentation/en-US/Red_Hat_JBoss_Fuse/6.1/html/Tooling_User_Guide/newCamelTestCase.html

it can help to implement it in Forge.
